Monika After Story/game/definitions.rpy

```diff
@@ -5,6 +5,102 @@ define config.developer = False #This is the flag for Developer tools
 init 1 python:
     persistent.steam = "steamapps" in config.basedir.lower()
 
+### Overrides of core renpy things
+python early in mas_overrides:
+    from store import *
+
+    def dummy(*args, **kwargs):
+        """
+        Dummy function that does nothing
+        """
+        return
+
+    # clear this so no more traceback. We expect node loops anyway
+    renpy.execution.check_infinite_loop = dummy
+
+    def get_field_override(self, field_name, args, kwargs):
+        """
+        Originally this method returns objects by references
+        This override allows us to eval functions when appropriate, e.g. "Text [myfunc(arg1)]."
+        and use negative indexes for iterables, e.g. "Text [_list[-2]]."
+
+        IN:
+            field_name - the reference to the object
+            args - not sure, but renpy doesn't use it (passes in an empty tuple)
+            kwargs - the store modules where renpy will look for the object
+
+        OUT:
+            tuple of the object and its key
+        """
+        # if it's a function, we eval it
+        if "(" in field_name:
+            # id to split the string
+            split_index = field_name.index("(")
```

Comment:
Is `field_name` a string? If so, just use `partition` since you are splitting a string into two parts.
```python
func_name, paren, args = field_name.partition("(")
```

Reply:
Shall I do the same for this line too?
```python
func_store_name, func_name = func_name.split(".", 1)
```

Reply:
sure